File: BinarySearch.java
import javax.swing.*;

public class BinarySearch {
    public static void main(String[] args) {

        int[] array = {0,1,2,3,4,5,6,7};

        int index = BinarySearch(array , 0);
        //System.out.println(array.length);

        if(index == -1){
            System.out.println("we couldn't find the value");
        }
        else{
            System.out.println("we found your value ");
        }
    }

    public static int BinarySearch(int[] array , int target){

        int low = 0;
        int high = array.length - 1;

        while(low <= high){

            int middle = low +(high - low)/2;

            System.out.println("middle: " + middle);
            int middleValue = array[middle];

            if(target > middleValue){
                low = middle + 1;
            }

            else if (target < middleValue ){
                high = middle - 1;
            }
            else {
                return middleValue;
            }
        }
        return -1;
    }
}

File: dsa-javascript/binarySearch.js
function binarySearch(array, target) {
    var low = 0;
    var high = array.length - 1;
    while (low <= high) {
        var middle = Math.floor(low + (high - low) / 2);
        console.log("Middle value is ".concat(middle, " and its associated value is ").concat(array[middle]));
        var middelValue = array[middle]; //to explicitly tell typescript compiler that this value will not be null at any point of 
        if (middelValue > target) {
            high = middle - 1;
        }
        else if (middelValue < target) {
            low = middle + 1;
        }
        else {
            return middelValue;
        }
    }
    return -1;
}
var array = [34, 56, 57, 90, 97, 98, 99, 102, 103, 104, 190];
var result = binarySearch(array, 9);
result === -1 ? console.log("The target value could not be found")
    : console.log("Result was found and the result is located on the array position of ".concat(result));
